apm-agent-plugins/apm-vertx/apm-vertx3-plugin/src/main/java/co/elastic/apm/agent/vertx/v3/web/WebHelper.java

```diff
@@ -72,22 +76,22 @@ public Transaction setRouteBasedNameForCurrentTransaction(RoutingContext routing
         return transaction;
     }
 
-    public void removeTransactionFromContext(HttpServerRequest request) {
-        if (request.getClass().getName().equals("io.vertx.ext.web.impl.HttpServerRequestWrapper")) {
-            request = request.endHandler(noopHandler);
-        }
-        requestTransactionMap.remove(request);
+    @Nullable
+    public Transaction removeTransactionFromContext(HttpServerRequest request) {
+        return requestTransactionMap.remove(request);
```

Comment:
Why did you remove the handling of the wrapped request case?
The purpose of that was that we never have the a wrapped request object (`io.vertx.ext.web.impl.HttpServerRequestWrapper`) as a key in our context-map.

Reply:
We now instrument the `io.vertx.core.http.impl.HttpServerRequestImpl#handleEnd` and `io.vertx.core.http.impl.Http2ServerRequestImpl#handleEnd` so there is no need for that

Reply:
If these events are not guaranteed to run, we have a bigger problem than leftover in the map - we won't end the transaction, so we can't use this approach